runtime/compiler/control/J9Recompilation.cpp

```diff
@@ -126,10 +126,18 @@ J9::Recompilation::setupMethodInfo()
 
    // TR_Controller is being used and has provided a compilation plan.  Use it to setup the compilation
    //
+   // JITServer does not support JitProfiling
+   bool performProfiling = (optimizationPlan->insertInstrumentation() != 0);
```

Comment:
I scanned the code and as far as I can tell `optimizationPlan->insertInstrumentation()` is false if `getOption(TR_DisableProfiling)` is true. Thus, we don't need the check on JITServer::SERVER or JITServer::CLIENT